source/server/listener_manager_impl.cc

```diff
@@ -965,11 +965,11 @@ Network::DrainableFilterChainSharedPtr ListenerFilterChainFactoryBuilder::buildF
 void ListenerManagerImpl::setNewOrDrainingSocketFactory(
     const std::string& name, const envoy::config::core::v3::Address& proto_address,
     ListenerImpl& listener) {
-  // For listeners that do not bind or listeners that do not bind to port 0 we must check to make
+  // For listeners that do not bind to port 0 we must check to make
   // sure we are not duplicating the address. This avoids ambiguity about which non-binding
   // listener is used or even worse for the binding to port != 0 and reuse port case multiple
   // different listeners receiving connections destined for the same port.
-  if ((!listener.bindToPort() || listener.config().address().socket_address().port_value() != 0) &&
+  if (listener.config().address().socket_address().port_value() != 0 &&
```

Comment:
This seems like an unrelated change? Why are we no longer checking to see if it's not being bound?

Reply:
This is due to the linux won't reject both ::FFFF:192.168.0.1 and 192.168.0.1, then we have to check that by Envoy.

Reply:
But shouldn't the not binding check be OK? I don't see how that's related to this PR. Can you explain further? Sorry if I'm missing something trivial.

Reply:
The issue is for the case of two listeners `::FFFF:192.168.0.1` and `192.168.0.1` added. On linux, two listeners were added successful, but the request always go to the `192.168.0.1`. On windows, Those tow listeners will be rejected by OS.

So for windows, it is ok, the OS will reject it directly. but on linux, the OS accept those two addresses are listening at same time, but actually, it is meanful.

In envoy, we only check address is duplicated or not when bind_to_port is `false` and non-zero port. So it removed `bind_to_port` check here.